onboarding/source/amb_light_service.c

```diff
@@ -34,20 +44,78 @@ static void lightServiceTask(void * pvParameters);
 obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here. Return error code if task/queue was not created successfully.
+    if (lightServiceTaskHandle == NULL) 
+    {
+        // Create light service task
+        lightServiceTaskHandle = xTaskCreateStatic( lightServiceTask,             /* Function that implements the task. */
+                                                    LIGHT_SERVICE_NAME,            /* Text name for the task. */
+                                                    LIGHT_SERVICE_STACK_SIZE,      /* Stack size in words, not bytes. */
+                                                    NULL,                       /* Parameter passed into the task. */
+                                                    LIGHT_SERVICE_PRIORITY,        /* Priority at which the task is created. */
+                                                    lightServiceTaskStack,        /* Array to use as the task's stack. */
+                                                    &lightServiceTaskBuffer);     /* Used to hold the task's data structure */
+    }
+
+    if (lightServiceTaskHandle == NULL)
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    
+    if (lightServiceQueueHandle == NULL) 
+    {
+        lightServiceQueueHandle = xQueueCreateStatic ( QUEUE_LENGTH,
+                                                       ITEM_SIZE,
+                                                       lightServiceQueueStorage,
+                                                       &lightServiceQueue);
+    }
+
+    if(lightServiceQueueHandle == NULL)
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+
+    return OBC_ERR_CODE_SUCCESS;
 
     /* USER CODE END */
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    if (lightServiceQueueHandle != NULL)
+    {
+        light_event_t event;
+        while (1)
+        {
+            if (xQueueReceive (lightServiceQueueHandle, 
+                                &event,
+                                ( TickType_t) 0) == pdTRUE && event == MEASURE_LIGHT)
```

Comment:
This is a little sketchy since `event` gets updated in the first part of the condition. I.e. changing this to `if (event == MEASURE_LIGHT && QueueReceive (lightServiceQueueHandle, &event, ( TickType_t) 0))` would yield different results from what you have. Also, this wouldn't scale well if we added new events, not just `MEASURE_LIGHT`. 

Instead, add a separate if statement to check if `event == MEASURE_LIGHT`